### PR TITLE
ToggleGroupControl: Improve stories for Docs view

### PR DESCRIPTION
**Pull request by mirka**
## What?

- Populate the props table with TypeScript data.
- Refactor the stories to use Controls instead of Knobs (#35665).
- Merge the WithAriaLabel story into the WithTooltip story.
- Remove the WithReset story since it doesn't serve a clear purpose anymore.
- Minor doc improvements.

## Why?

To improve DevEx and documentation.

## Testing Instructions

1. `npm run storybook:dev` and see the ToggleGroupControl page in Docs view.

### CODE REVIEW

File: packages/components/src/toggle-group-control/stories/index.tsx [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+	ToggleGroupControlProps,
+} from '../types';
+
+const meta: ComponentMeta< typeof ToggleGroupControl > = {
+	component: ToggleGroupControl,
+	title: 'Components (Experimental)/ToggleGroupControl',
+	subcomponents: { ToggleGroupControlOption, ToggleGroupControlOptionIcon },
+	argTypes: {
+		// TODO: Should this component be non-polymorphic?
+		as: { control: { type: null } },
```
Comment (mirka) [inline packages/components/src/toggle-group-control/stories/index.tsx]: @ciampo Thoughts? I don't think it makes sense for this to be polymorphic.

Comment (ciampo) [reply packages/components/src/toggle-group-control/stories/index.tsx]: Makes sense — this component has quite strong semantics, afterall

File: packages/components/src/toggle-group-control/stories/index.tsx [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+	ToggleGroupControlOptionIconProps,
+	ToggleGroupControlProps,
+} from '../types';
+
+const meta: ComponentMeta< typeof ToggleGroupControl > = {
+	component: ToggleGroupControl,
+	title: 'Components (Experimental)/ToggleGroupControl',
+	subcomponents: { ToggleGroupControlOption, ToggleGroupControlOptionIcon },
+	argTypes: {
+		// TODO: Should this component be non-polymorphic?
```
Comment (ciampo) [inline packages/components/src/toggle-group-control/stories/index.tsx]: 🧹

File: packages/components/src/toggle-group-control/stories/index.tsx [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+			'aria-label': 'Ascending',
+			showTooltip: true,
+		},
+		{
+			value: 'desc',
+			label: 'Z→A',
+			'aria-label': 'Descending',
+			showTooltip: true,
+		},
+	].map( mapPropsToOptionComponent ),
```
Comment (ciampo) [inline packages/components/src/toggle-group-control/stories/index.tsx:109]: Neat!

**Review by ciampo — APPROVED** [review]
Great work 🚀

**Pull request merged**